app/models/cart.rb

```diff
@@ -238,4 +220,25 @@ def fields_for_display
       EXCLUDE_FIELDS_FROM_DISPLAY.include? key}.map{ |key,value,label|
       [label, value] }
   end
+  def client
+    # TODO set the origin
+    if self.origin.blank?
+      "gsa_advantage"
+    else
+      self.origin
+    end
+  end
+  # @todo - the method name (e.g. :external_id) should live on a "client"
+  # model
+  def public_identifier
+    self.external_id
+  end
+  def total_price
+    case self.getProp('origin')
+    when 'gsa18f'
```

Comment:
Mind changing to `if self.origin == 'gsa18f'`? Maybe better yet, add a `#gsa18f?` method.